src/Dialog/Dialog.tsx

```diff
@@ -309,11 +404,18 @@ const _Dialog = React.forwardRef<HTMLDivElement, React.PropsWithChildren<DialogP
   const header = (renderHeader ?? DefaultHeader)(defaultedProps)
   const body = (renderBody ?? DefaultBody)(defaultedProps)
   const footer = (renderFooter ?? DefaultFooter)(defaultedProps)
+  const positionDataAttributes =
+    typeof position === 'string'
+      ? {'data-position-regular': position}
+      : Object.entries(position).reduce((acc: {[key: string]: string}, [key, value]) => {
+          acc[`data-position-${key}`] = value
+          return acc
+        }, {})
```

Comment:
Just wanted to share another alternative, they're totally doing the same thing but figured I'd share in case it was interesting.

```suggestion
  const positionDataAttributes =
    typeof position === 'string'
      ? {'data-position-regular': position}
      : Object.fromEntries(
          Object.entries(position).map(([key, value]) => {
            return [`data-position-${key}`, value]
          }),
        )

```